src/containers/User/AddUser/index.js

```diff
@@ -0,0 +1,168 @@
+import React from  'react';
+import {
+  Button,
+  Card,
+  Checkbox,
+  Col,
+  Form,
+  Input,
+  Row,
+  Select,
+  Switch,
+  Typography,
+} from 'antd';
+import { PlusOutlined } from '@ant-design/icons';
+import PropTypes from 'prop-types';
+
+import styles from './style.module.css';
+
+const { Title } = Typography;
+const { Option } = Select;
+const requiredMessage = 'Este campo é obrigatório!';
+
+const formRequireRules = [{
+  message: requiredMessage,
+  required: true,
+}]
+
+const AddUser = ({
+  allowAddTypeAccount,
+  allowCustomPermissions,
+  form,
+  goToAddTypeAccount,
+  handleAllowSetCustomPermissions,
+  handleOnTypeAccountChange,
+  handleSubmit,
+  permissions,
+  typeAccounts,
+}) => {
+    const renderCheckbox = ({
+      label,
+      name,
+      value,
+    }) => (
+      <Col
+        className={styles.checkBoxheight}
+        span={12}
+        key={name}
+      >
+        <Form.Item
+          name={name}
+          valuePropName="checked"
+        >
+          <Checkbox
+            disabled={!allowCustomPermissions}
+            value={value}
+          >
+            {label}
+          </Checkbox>
+        </Form.Item>
+      </Col>
+    )
+
+    const renderTypeAccountOptions = (typeAccountItem, index) => (
+      <Option
+        key={index}
+        value={typeAccountItem.typeName}
+      >
+        {typeAccountItem.typeName}
+      </Option>
+    )
+
+    return (
+      <div className={styles.addUserContainer}>
+        <Form form={form} onFinish={handleSubmit}>
+          <Row justify="center">
+            <Col>
+              <Title level={3}>Novo usuário</Title>
+            </Col>
+          </Row>
+          <Row gutter={[8, 8]}>
+            <Col span={12}>
+              <Form.Item
+                label="Usuário"
+                name="userName"
+                rules={formRequireRules}
+              >
+                <Input />
+              </Form.Item>
+            </Col>
+
+            <Col span={12}>
+              <Form.Item
+                label="Tipo de conta"
+                name="typeAccount"
+                rules={formRequireRules}
+              >
+                <Select
+                  onChange={handleOnTypeAccountChange}
+                  placeholder="Selecione o tipo de conta!"
+                  style={{
+                    width: '80%',
+                    marginRight: '6px',
+                  }}
+                >
+                  {typeAccounts && typeAccounts.map(renderTypeAccountOptions)}
+                </Select>
+                { allowAddTypeAccount && (
+                  <Button
+                    className="buttonadd-marca-produtos"
+                    onClick={goToAddTypeAccount}
+                    type="primary"
+                  >
+                    <PlusOutlined />
+                  </Button>
+                )}
+              </Form.Item>
+            </Col>
+          </Row>
+
+          <Form.Item label="Habilitar customização:">
```

Comment:
adicionar um prop name, caso contrário ao resetar o form o switch não terá seu valor alterado
```suggestion
          <Form.Item 
            label="Habilitar customização:"
            name="allowCustomPermissions"
            valuePropName="checked"
          >
```